agent/src/com/jonnyzzz/teamcity/virtual/run/docker/DockerContext.java

```diff
@@ -37,4 +37,30 @@ public String getImageName() throws RunBuildException {
     if (StringUtil.isEmptyOrSpaces(image)) throw new RunBuildException("Docker image is not specified");
     return image;
   }
+
+  public boolean isDockerServerWindowsBased() throws RunBuildException {
+    final String serverArch = myContext.getConfigParameters().get(VMConstants.DOCKER_HOST_OS_PROPERTY);
+    return serverArch.equals("windows");
+  }
+
+  @NotNull
+  public String getShellLocationInsideContainer() throws RunBuildException {
```

Comment:
How does `getShellLocation` and the method related to each other? Looks like a code duplicate